cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -596,6 +603,79 @@ mkConsensusTracers mbEKGDirect trSel verb tr nodeKern fStats = do
      when (isRollForward ev) $
        sendEKGDirectCounter ekgDirect "cardano.node.metrics.served.header.counter.int"
 
+traceBlockFetchClientMetrics
+  :: forall blk remotePeer.
+     ( )
+  => Maybe EKGDirect
+  -> STM.TVar (SMap.Map SlotNo NominalDiffTime)
+  -> STM.TVar Int64
+  -> STM.TVar Int64
+  -> STM.TVar Int64
+  -> Tracer IO (TraceLabelPeer remotePeer (TraceFetchClientState (Header blk)))
+  -> Tracer IO (TraceLabelPeer remotePeer (TraceFetchClientState (Header blk)))
+traceBlockFetchClientMetrics Nothing _ _ _ _ tracer = tracer
+traceBlockFetchClientMetrics (Just ekgDirect) slotMapVar cdf1sVar cdf2sVar cdf5sVar tracer = Tracer bfTracer
+
+  where
+
+    -- We track the cdf for 1s, 2s and 5s.
+    updateCDFCounter :: NominalDiffTime -> Int -> Int64 -> STM (Double, Double, Double)
+    updateCDFCounter delay size step = do
+
+      when (delay < 1) $ do
+        STM.modifyTVar' cdf1sVar (+ step)
+      when (delay < 2) $ do
+        STM.modifyTVar' cdf2sVar (+ step)
+      when (delay < 5) $ do
+        STM.modifyTVar' cdf5sVar (+ step)
+
```

Comment:
do we want to track >= 5s?

Reply:
All samples that are still > 5s are still stored in the map that tracks slotnumbers and delay so they are accounted for in the cdfs.